internal/terminal/browser.go

```diff
@@ -0,0 +1,22 @@
+package terminal
+
+import "github.com/pkg/browser"
+
+// go:generate mockgen -destination=internal/utils/test/mock/browser.go -package=mock github.com/10gen/realm-cli/internal/terminal Browser
```

Comment:
this is super :neat: but probably overkill...  since we already have the `terminal.UI` interface and its own (manual) mock for it, is it possible for us to just:

1) keep `OpenBrowser(url string) error` on the `UI` interface
2) remove the `browser Browser` field from the `terminal.ui` struct
3) call `browser.OpenURL(url)` in the impl of the `(*terminal.ui).OpenBrowser(string)` impl
4) add a `OpenURLFn func(url string) error` to the `mock.ui` definition in `internal/utils/test/mock/terminal_ui` (and leverage it the way we usually leverage "mock fns")

it seems like currently, the mock exists to determine if the browser was opened to a certain URL, and I think that manual mock fn could serve a similar purpose (without the addition of a new package and a `//go:generate`)

(btw, I **love** `go:generate` for a lot of the things, it hurts me here to suggest ripping it out... but I do think we can end up a bit simpler without it)

---

this likely has the added affect of not changing the constructor for a `terminal.NewUI` as well